modules/sdk-lib-mpc/src/tss/ecdsa/rangeproof.ts

```diff
@@ -104,6 +104,39 @@ export async function generateNtilde(bitlength = minModulusBitLength): Promise<D
   };
 }
 
+/**
+ * Generate non-interactive challenge for Ntilde DLOG proofs.
+ * @param {DeserializedNtilde} ntilde Ntilde, h1, h2 to generate the proofs for.
+ * @param {bigint[]} alpha array of alpha values from the Ntilde proofs.
+ * @returns {Buffer} buffer of bytes reperesenting te challenge.
+ */
+function generateNIZKChallenge(ntilde: DeserializedNtilde, alpha: bigint[]): Buffer {
+  if (alpha.length !== ITERATIONS) {
+    throw Error(`List of alpha values has to be of length ${ITERATIONS}`);
+  }
+  const modulus_bytes: number = Math.ceil(bitLength(ntilde.ntilde) / 8);
+  const modulus_bytes_buf: Buffer = bigIntToBufferBE(BigInt(modulus_bytes));
+  let msgToHash: Buffer = Buffer.concat([
+    modulus_bytes_buf,
+    bigIntToBufferBE(ntilde.h1, modulus_bytes),
```

Comment:
I don't think it's of any particular concern here, but `bigIntToBufferBE` pads up to `minBytes` if the input is small/short, however it doesn't truncate if input is longer.  Just something to keep in mind.